CartingService/Messaging/MessagingService.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.Text.Json;
+using Confluent.Kafka;
+using Microsoft.Extensions.Logging;
+
+public class MessagingService
+{
+    private readonly IItemService _itemService;
+    private readonly ILogger _logger;
+
+    public MessagingService(IItemService itemService, ILogger<MessagingService> logger)
+    {
+        _itemService = itemService;
+        _logger = logger;
+    }
+
+    public async Task Consume(string server, string groupId, CancellationToken cancellationToken)
+    {
+        using var consumer = new ConsumerBuilder<Ignore, string>(new ConsumerConfig { GroupId = groupId, BootstrapServers = server }).Build();
+
+        consumer.Subscribe(nameof(ItemUpdatedMessage));
+        while (!cancellationToken.IsCancellationRequested)
+        {
+            var message = consumer.Consume(cancellationToken);
+            var messageHandled = false;
+            while (!messageHandled)
+            {
+                try
+                {
+                    if (message.Topic == nameof(ItemUpdatedMessage))
+                    {
+                        await _itemService.HandleItemUpdated(JsonSerializer.Deserialize<ItemUpdatedMessage>(message.Message.Value)!);
```

Comment:
What very important restriction was introduced here? Let's imagine that the message that is handled should create a new record in the DB, will this approach work correctly here?